### ISSUE TITLE
OSX local dev improvements 

### ISSUE THREAD

**Issue by adamdecaf** [mozilla/tls-observatory#316]
From the few larger changes I've submitted there are a few local changes I keep around. Are there thoughts on adding them? I can probably fit them all in so linux and osx dev work at the same time. 

This can likely be changed with a `if` inside the makefile. 
```
modified   Makefile
@@ -9,7 +9,7 @@ BUILDREV	:= $(BUILDDATE)+$(BUILDREF).$(BUILDENV)
 
 # Supported OSes: linux darwin windows
 # Supported ARCHes: 386 amd64
-OS			:= linux
+OS			:= darwin
```

I believe this is resolved by upgrading the vendored `crypto/x509` fork in this project, but I don't know if there are any diffs applied to our version. See https://github.com/zmap/zcrypto/issues/106 and https://github.com/google/certificate-transparency-go/commit/391726f8973ddb3521ab1a462bd55a35c6f67f0b 
```
modified   certificate/constraints/vendor/constraintcrypto/x509/root_cgo_darwin.go
@@ -157,7 +157,7 @@ import (
 func loadSystemRoots() (*CertPool, error) {
 	roots := NewCertPool()
 
-	var data C.CFDataRef = nil
+	var data C.CFDataRef
```

Lastly, I've mucked around with the `conf/*` files, but I don't think that's important to commit upstream. Maybe just exposing the `postgres` port onto the docker host.

**Comment by jvehent** [comment]
I'm in favor of making development on macos easier, that said:

> This can likely be changed with a if inside the makefile.

You can override this on the command line by setting `OS=darwin make`

> I believe this is resolved by upgrading the vendored crypto/x509 fork in this project, but I don't know if there are any diffs applied to our version.

The one under `certificate/constraints/vendor/constraintcrypto/x509` has been customized and would need to be rebased.

> Lastly, I've mucked around with the conf/* files

You can easily point to a customized conf file with `-c`.

**Comment by jvehent** [comment]
not at all. make a PR with updated deps and we'll merge it if CI is happy
